proposals/nnnn-implicit-open-existentials.md

```diff
@@ -0,0 +1,387 @@
+
+```swift
+protocol P {
+  associatedtype A
+  func getA() -> A
+}
+
+func takeP<T: P>(_ value: T) { }
+
+func test(p: any P) {
+  takeP(p) // error: protocol 'P' as a type cannot conform to itself
+}
+```
+
+This interaction with the generics system makes existentials a bit of a trap in Swift: it's easy to go from generics to existentials, but once you have an existential it is very hard to go back to using it generically. At worst, you need to go back through many levels of functions, changing their parameters or results from `any P` to being generic over `P`, or writing a custom [type eraser](https://www.swiftbysundell.com/articles/different-flavors-of-type-erasure-in-swift/). 
+
+This proposal addresses this existential trap by allowing one to "open" an existential value, binding a generic parameter to its underlying type. Doing so allows us to call a generic function with an existential value, such that the generic function operations on the underlying value of the existential rather than on the existential box itself, making it possible to get out of the existential trap without major refactoring.
+
+Swift-evolution thread: [Discussion thread topic for that proposal](https://forums.swift.org/)
+
+## Motivation
+
+Swift's existentials are a powerful tool for working with a dynamic set of types dynamically, but their inability to interoperate well with Swift's generics is limiting and often leads developers into an existential "trap" where one has a value of existential type but cannot access necessary generic operations on that value.  Designs that fall into the existential trap aren't necessarily bad designs, but the workarounds needed for them involve a lot of refactoring (e.g., turning everything generic) or building custom type eraser types. These involve advanced language features and techniques, resulting in a significant amount of boilerplate as well as creating a steep learning curve for the language. Smoothing out this interaction between existentials and generics can simplify Swift code and make the language more approachable.
+
+## Proposed solution
+
+To make it easier to move from existentials back to the more strongly-typed generics, we propose to implicitly *open* an existential value when it is passed to a parameter of generic type. In such cases, the generic argument refers to the *underlying* type of the existential value rather than the existential "box". Let's start with a protocol `Costume` that involves `Self` requirements, and write a generic function that checks some property of a costume:
+
+```swift
+protocol Costume {
+  func withBells() -> Self
+  func hasSameAdornments(as other: Self) -> Bool
+}
+
+// Okay: generic function to check whether adding bells changes anything
+func hasBells<C: Costume>(_ costume: Costume) -> Bool {
```

Comment:
I bet this should be:
```suggestion
func hasBells<C: Costume>(_ costume: C) -> Bool {
```

Still, since SE-0341 is accepted now, does it make sense to simplify the line as:
```suggestion
func hasBells(_ costume: some Costume) -> Bool {
```